velox/functions/sparksql/specialforms/GetStructField.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/sparksql/specialforms/GetStructField.h"
+#include "velox/expression/ConstantExpr.h"
+
+namespace facebook::velox::functions::sparksql {
+namespace {
+
+class GetStructFieldFunction : public exec::VectorFunction {
+ public:
+  GetStructFieldFunction(int32_t ordinal) : ordinal_(ordinal) {}
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& resultType,
+      exec::EvalCtx& context,
+      VectorPtr& result) const override {
+    exec::LocalDecodedVector decoded(context, *args[0], rows);
+    auto rowData = decoded->base()->as<RowVector>();
+    if (decoded->isIdentityMapping()) {
+      result = rowData->childAt(ordinal_);
+    } else {
+      result =
+          decoded->wrap(rowData->childAt(ordinal_), *args[0], decoded->size());
```

Comment:
Would you please add tests for different encodings?

Reply:
added